ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/components/CreateOnDemandReportModal.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useState } from 'react';
+import { Modal, Button, Alert, Flex, FlexItem } from '@patternfly/react-core';
+
+type Message = {
+    type: 'success' | 'error';
+    value: string;
+} | null;
+
+const defaultMessage: Message = null;
+
+export type CreateOnDemandReportModalProps = {
+    isOpen: boolean;
+    setIsOpen: (value: boolean) => void;
+    query: string;
+    areaOfConcern: string;
+};
+
+function CreateOnDemandReportModal({
+    isOpen,
+    setIsOpen,
+    // @TODO: Will use "query" in a future PR
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    query,
+    // @TODO: Will use "areaOfConcern" in a future PR
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    areaOfConcern,
+}: CreateOnDemandReportModalProps) {
+    const [isTriggeringReportGeneration, setIsTriggeringReportGeneration] = useState(false);
+    const [message, setMessage] = useState(defaultMessage);
+
+    const handleModalToggle = () => {
+        setIsOpen(!isOpen);
+        setMessage(null);
+    };
+
+    const triggerOnDemandReportGeneration = () => {
+        setIsTriggeringReportGeneration(true);
+        // @TODO: Do an actual API call. This is just for demonstration.
+        const promise = new Promise((resolve, reject) => {
+            setTimeout(() => {
+                if (Math.random() < 0.5) {
+                    resolve({ reportID: '123456789' });
+                } else {
+                    reject('Something went wrong. Please contact support for assistance.');
```

Comment:
@mansursyed Let me know if there is a more appropriate error message we'd like to display.